### PR TITLE
Support propagating frameworks via objc/swift_library data attribute.

### PR DESCRIPTION
**Pull request by keith**
Adds support to propagate Apple frameworks (ie. `ios_framework`,
`tvos_framework`) through the `data` attribute from transitive
libraries (ie. `objc_library` and `swift_library`) using the
resource aspect.

Currently, Apple frameworks (`*_framework` targets) are supported
using the `runtime_deps` attribute for `objc_library` targets
through the framework provider aspect (support was added at
https://github.com/bazelbuild/rules_apple/commit/1199da04e51c273f5509a9153c32d4d8dd75b336 and https://github.com/bazelbuild/rules_apple/commit/889131aca743079cdff68cba27b4987ca301f9bb). This support will be deprecated
on a follow-up change.

PiperOrigin-RevId: 513139752
(cherry picked from commit f711be032bd62fa13e866607bfb238e1988aca89)

### CODE REVIEW

**Comment by stravinskii** [comment]
FYI - [644faceabc61d7fe46eaed56c8da6ca95766e38a](https://github.com/bazelbuild/rules_apple/commit/644faceabc61d7fe46eaed56c8da6ca95766e38a) fixes support for tree artifacts.

**Comment by keith** [comment]
bump

File: apple/internal/resources.bzl [hunk 1]
```diff
@@ -382,13 +402,15 @@ def _bucketize_typed(resources, bucket_type, *, owner = None, parent_dir_param =
...
 
     Args:
-        resources: List of resources to place in bucket_type.
         bucket_type: The AppleResourceInfo field under which to collect the resources.
         owner: An optional string that has a unique identifier to the target that should own the
             resources. If an owner should be passed, it's usually equal to `str(ctx.label)`.
         parent_dir_param: Either a string/None or a struct used to calculate the value of
             parent_dir for each resource. If it is a struct, it will be considered a partial
             context, and will be invoked with partial.call().
+        resources: List of resources to place in bucket_type or Dictionary of resources keyed by
```
Comment (brentleyjones) [inline apple/internal/resources.bzl:411]: Odd this moved down.

**Review by brentleyjones — APPROVED** [review]

File: apple/internal/resources.bzl (hunk 1 above)
Comment (keith) [reply apple/internal/resources.bzl:411]: hrm yea seems like buildifier doesn't care about the ordering 🤔

**Pull request merged**